packages/core/test/actorLogic.test.ts

```diff
@@ -232,6 +232,46 @@ describe('promise logic (fromPromise)', () => {
 
     createActor(promiseLogic).start();
   });
+
+  it('should abort when stopping', (done) => {
+    let signal: AbortSignal;
+    const promiseLogic = fromPromise((ctx) => {
+      signal = ctx.signal;
+      return new Promise((res) => {
+        setTimeout(() => res(42), 1000);
+      });
```

Comment:
could we rewrite those test cases in a way that would show more real-life usage? `signal` shouldn't have to be leaked to the outer scope to get a meaningful test case for this

Reply:
[`c06b029` (#4832)](https://github.com/statelyai/xstate/pull/4832/commits/c06b029019d1aac6d9d66166a5a940c4f49661f8)